### PR TITLE
catch region outdated error and fix test bug

### PR DESCRIPTION
**Pull request by c4pt0r**

### CODE REVIEW

**Comment by c4pt0r** [comment]
@disksing @siddontang PTAL

File: themis_rpc.go [hunk 1]
```diff
@@ -202,7 +202,10 @@ func (rpc *themisRPC) commitRow(tbl, row []byte, mutations []*columnMutation,
 	}
 	ok := res.GetResult()
 	if !ok {
-		return errors.New(fmt.Sprintf("commit failed, tbl: %s row: %s ts: %d", tbl, row, commitTs))
+		if primaryOffset == -1 {
```
Comment (siddontang) [inline themis_rpc.go]: use %q for row

Comment (c4pt0r) [inline themis_rpc.go]: fixed

**Comment by c4pt0r** [comment]
@siddontang PTAL

File: themis_txn.go [hunk 2]
```diff
@@ -391,9 +403,13 @@ func (txn *themisTxn) tryToCleanLock(lock Lock) error {
 				// if this transction has been committed
 				log.Info("txn has been committed, ts:", commitTs, "prewriteTs:", pl.Timestamp())
 				// commit secondary row
-				return txn.commitSecondaryAndCleanLock(lock.(*themisSecondaryLock), commitTs)
+				err := txn.commitSecondaryAndCleanLock(lock.(*themisSecondaryLock), commitTs)
+				if err != nil {
+					log.Error(err)
```
Comment (siddontang) [inline themis_txn.go]: need to log Error here? following will return this error.

**Comment by siddontang** [comment]
Rest LGTM

File: themis_txn.go [hunk 3]
```diff
@@ -155,17 +155,27 @@ func (txn *themisTxn) Commit() error {
 	txn.selectPrimaryAndSecondaries()
 	err := txn.prewritePrimary()
 	if err != nil {
-		return errors.Trace(err)
+		// no need to check wrong region here, hbase client will retry when
+		// occurs single row NotInRegion error.
+		log.Error(err)
```
Comment (qiuyesuifeng) [inline themis_txn.go]: Any need to use ErrorStack?

File: themis_txn.go [hunk 4]
```diff
@@ -155,17 +155,27 @@ func (txn *themisTxn) Commit() error {
...
+		return kv.ErrRetryable
 	}
 
 	err = txn.prewriteSecondary()
 	if err != nil {
-		return errors.Trace(err)
+		if isWrongRegionErr(err) {
+			log.Warn("region info outdated")
+			// reset hbase client buffered region info
+			txn.client.CleanAllRegionCache()
```
Comment (qiuyesuifeng) [inline themis_txn.go]: Will this function return error?

**Comment by qiuyesuifeng** [comment]
LGTM

File: themis_txn.go (hunk 4 above)
Comment (c4pt0r) [inline themis_txn.go]: no

**Pull request merged**